src/compiler/checker.ts

```diff
@@ -8503,15 +8577,48 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                     return visited;
                 }
-
-                if (isEntityName(node) || isEntityNameExpression(node)) {
-                    if (isDeclarationName(node)) {
-                        return node;
+                if (isTypeQueryNode(node)) {
+                    const { introducesError, node: exprName } = trackExistingEntityName(node.exprName, context);
+                    if (introducesError) {
+                        const serializedName = serializeTypeName(context, node.exprName, /*isTypeOf*/ true);
+                        if (serializedName) {
+                            serializeExistingTypeNode(context, node);

```

Comment:
Serialize for side effects? Probably warrants a comment explaining why. And being inside a `void` expression (which is just a personal preference on my part).

Reply:
Was left over from some debugging. removed.